client/cli/src/commands/vanity.rs

```diff
@@ -16,77 +16,92 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-use super::{PublicOf, PublicT, Crypto};
-use sp_core::Pair;
-use rand::{rngs::OsRng, RngCore};
+//! implementation of the `vanity` subcommand
 
-fn good_waypoint(done: u64) -> u64 {
-	match done {
-		0..=1_000_000 => 100_000,
-		0..=10_000_000 => 1_000_000,
-		0..=100_000_000 => 10_000_000,
-		_ => 100_000_000,
-	}
+use crate::{
+	error, utils, CliConfiguration, with_crypto_scheme,
+	CryptoSchemeFlag, NetworkSchemeFlag, OutputTypeFlag, SharedParams,
+};
+use sp_core::crypto::Ss58Codec;
+use structopt::StructOpt;
+use rand::{rngs::OsRng, RngCore};
+use sp_runtime::traits::IdentifyAccount;
+
+/// The `vanity` command
+#[derive(Debug, StructOpt)]
+#[structopt(
+	name = "vanity",
+	about = "Generate a seed that provides a vanity address"
+)]
+pub struct VanityCmd {
+	/// Number of keys to generate
+	#[structopt(long, short)]
+	number: String,
+
+	/// Desired pattern
+	#[structopt(long)]
+	pattern: Option<String>,
```

Comment:
Can this be a `String` instead?

Reply:
If the `--pattern` is not required then it must be an `Option<String>`. If it is required, then it must be a `String`.